lib/Dialect/FIRRTL/Transforms/BlackBoxReader.cpp

```diff
@@ -46,6 +46,21 @@ static void appendPossiblyAbsolutePath(SmallVectorImpl<char> &base,
   }
 }
 
+SmallVector<StringRef> splitString(std::string &s, char delim) {
+  SmallVector<StringRef> retval;
+  size_t start = 0U;
+  size_t end;
+  do {
+    end = s.find(delim, start);
+    if (end == std::string::npos)
+      break;
+    retval.emplace_back(&s.data()[start], end - start);
+    start = end + 1;
+  } while (1);
+  retval.emplace_back(&s.data()[start], s.size() - start);
+  return retval;
+}
+
```

Comment:
Does [this](https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/ADT/StringRef.h#L799) work?

Reply:
Agreed, I'd use a split member on StringRef